File: rendezvous_server.py
# rendezvous_server.py
from flask import Flask, request, jsonify
from flask_cors import CORS
import ssl

app = Flask(__name__)
CORS(app)

peers = {}

@app.route('/register', methods=['POST'])
def register():
    data = request.json
    peer_id = data['peer_id']
    address = data['address']
    peers[peer_id] = address
    return jsonify({"status": "registered", "peer_id": peer_id})

@app.route('/get_peer', methods=['POST'])
def get_peer():
    data = request.json
    requested_id = data['requested_id']
    requester_id = data['requester_id']
    if requested_id in peers:
        return jsonify({
            "status": "success",
            "peer_address": peers[requested_id],
            "requester_address": peers[requester_id]
        })
    else:
        return jsonify({"status": "not_found"})
    
@app.route('/', methods=['GET'])
def get_peers():
    print(peers)
    return jsonify(peers)

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=6000)

File: rendezvous_server.js
// rendezvous_server.js
const express = require('express');
const bodyParser = require('body-parser');

const app = express();
const port = 8000; // You can change this port as needed

// Middleware
app.use(bodyParser.json());

// Store peer information
const peers = {};

// Register a peer
app.post('/register', (req, res) => {
    const { peer_id, address } = req.body;
    
    if (!peer_id || !address) {
        return res.status(400).json({ error: 'Missing peer_id or address' });
    }

    peers[peer_id] = address;
    res.json({ status: 'registered', peer_id });
});

// Get peer information
app.post('/get_peer', (req, res) => {
    const { requested_id, requester_id } = req.body;

    if (!requested_id || !requester_id) {
        return res.status(400).json({ error: 'Missing requested_id or requester_id' });
    }

    if (!(requested_id in peers)) {
        return res.status(404).json({ error: 'Requested peer not found' });
    }

    if (!(requester_id in peers)) {
        return res.status(404).json({ error: 'Requester not registered' });
    }

    res.json({
        status: 'success',
        peer_address: peers[requested_id],
        requester_address: peers[requester_id]
    });
});

// Start the server
app.listen(port, () => {
    console.log(`Rendezvous server running on http://localhost:${port}`);
});